### PR TITLE
Add @with_jurisdiction_access route decorator

### PR DESCRIPTION
**Pull request by jonahkagan**
**Description**

Adds a route decorator `@with_jurisdiction_access`. This decorator requires jurisdiction admin (JA) authorization, checks that it matches the given election/jurisdiction ids, and loads the election and jurisdiction objects.

Also modifies `@with_election_access` to not take a user type and only accept audit admin (AA) authorization.

To simplify permissions for the time being, AAs and JAs will not share any routes. JA routes will always have a `jurisdiction_id` in the path so we can easily check that they are trying to access the jurisdiction for which they are authorized.

**Testing**

Lots of new tests here, trying to cover every case of incorrect authorization - could use a good review to make sure I'm not missing something.

Here's a summary:
 - test_with_election_access_audit_admin
 - test_with_election_access_wrong_org
 - test_with_election_access_not_found
 - test_with_election_access_jurisdiction_admin
 - test_with_election_access_anonymous_user
 - test_with_jurisdiction_access_jurisdiction_admin
 - test_with_jurisdiction_access_wrong_org
 - test_with_jurisdiction_access_election_not_found
 - test_with_jurisdiction_access_jurisdiction_not_found
 - test_with_jurisdiction_access_audit_admin
 - test_with_jurisdiction_access_anonymous_user

**Progress**

Ready for review.

### CODE REVIEW

File: tests/helpers.py [hunk 1]
```diff
@@ -24,46 +32,67 @@ def put_json(client: FlaskClient, url: str, obj) -> Any:
...
-        },
+    election = Election(
+        id=str(uuid.uuid4()),
+        audit_name=audit_name,
+        organization_id=organization_id,
+        is_multi_jurisdiction=is_multi_jurisdiction,
     )
-    return json.loads(rv.data)["electionId"]
+    db.session.add(election)
+    db.session.commit()
```
Comment (benadida) [inline tests/helpers.py:94]: why not keep the more functional test of calling `/election/new`? I could see unit tests drifting a bit in coverage, and then we miss an edge case.

File: tests/routes_tests/test_auth.py [hunk 2]
```diff
@@ -103,51 +126,224 @@ def test_jurisdictionadmin_start(client: FlaskClient):
...
-        id=str(uuid.uuid4()), email="bar@example.com", external_id="bar@example.com"
-    )
-    j_admin = JurisdictionAdministration(user_id=u.id, jurisdiction_id=jurisdiction.id)
-
-    db.session.add(election)
-    db.session.add(jurisdiction)
-    db.session.add(u)
-    db.session.add(j_admin)
-    db.session.commit()
-
```
Comment (benadida) [inline tests/routes_tests/test_auth.py:127]: did the behavior change here where you're no longer expecting a user in the table coming from back Auth0? I think that's worth a longer discussion if so.

File: tests/routes_tests/test_auth.py [hunk 3]
```diff
@@ -103,51 +126,224 @@ def test_jurisdictionadmin_start(client: FlaskClient):
...
+
+
+def test_with_election_access_audit_admin(client: FlaskClient, election_id: str):
+    set_logged_in_user(client, UserType.AUDIT_ADMIN, AA_EMAIL)
+    rv = client.get(f"/election/{election_id}/test_auth")
+    assert rv.status_code == 200
+    assert json.loads(rv.data) == election_id
+
+
+def test_with_election_access_wrong_org(
```
Comment (benadida) [inline tests/routes_tests/test_auth.py:171]: 😍

File: tests/helpers.py (hunk 1 above)
Comment (jonahkagan) [reply tests/helpers.py:94]: I made this change for two reasons:
- With the way it was before, I was gonna have to first create an admin and then set them as the logged in user in the session in order to access the endpoint. (For some specific test I was working on.)
- All of the other helpers do it this way, so it seemed consistent.

But I do hear your point that in general it's good to test everything only using the external interface, so I could be convinced to change it back.

File: tests/routes_tests/test_auth.py (hunk 2 above)
Comment (jonahkagan) [reply tests/routes_tests/test_auth.py:127]: I just removed this because the same code is now executed by the `jurisdiction_id` fixture (and the chain of fixtures it depends on).

Hopefully there's no behavior change. But maybe I missed something because I haven't actually worked with the auth0 integration yet.

File: tests/helpers.py (hunk 1 above)
Comment (benadida) [reply tests/helpers.py:94]: I would recommend keeping this as an API call, but it's not a hill I need to die on, so happy if you make the call you think is best.

File: tests/routes_tests/test_auth.py (hunk 2 above)
Comment (benadida) [reply tests/routes_tests/test_auth.py:127]: got it, so no change in behavior, just a different place that creates the local admin jurisdiction user? Ok great.

File: tests/helpers.py (hunk 1 above)
Comment (jonahkagan) [reply tests/helpers.py]: Cool, changed back to API call
